pkg/test/cel/httproute_test.go

```diff
@@ -1391,3 +1391,105 @@ func validateHTTPRoute(t *testing.T, route *gatewayv1b1.HTTPRoute, wantErrors []
 		t.Errorf("Unexpected response while creating HTTPRoute %q; got err=\n%v\n;missing strings within error=%q", fmt.Sprintf("%v/%v", route.Namespace, route.Name), err, missingErrorStrings)
 	}
 }
+
+func toDuration(durationString string) *gatewayv1b1.Duration {
+	return (*gatewayv1b1.Duration)(&durationString)
+}
+
+func TestHTTPRouteTimeouts(t *testing.T) {
+	tests := []struct {
+		name       string
+		wantErrors []string
+		rules      []gatewayv1b1.HTTPRouteRule
+	}{
+		{
+			name:       "invalid timeout unit us is not supported",
+			wantErrors: []string{"Invalid value: \"100us\": spec.rules[0].timeouts.request in body should match '^([0-9]{1,5}(h|m|s|ms)){1,4}$'"},
+			rules: []gatewayv1b1.HTTPRouteRule{
+				{
+					Timeouts: &gatewayv1b1.HTTPRouteTimeouts{
+						Request: toDuration("100us"),
+					},
+				},
+			},
+		},
+		{
+			name:       "invalid timeout unit ns is not supported",
+			wantErrors: []string{"Invalid value: \"500ns\": spec.rules[0].timeouts.request in body should match '^([0-9]{1,5}(h|m|s|ms)){1,4}$'"},
+			rules: []gatewayv1b1.HTTPRouteRule{
+				{
+					Timeouts: &gatewayv1b1.HTTPRouteTimeouts{
+						Request: toDuration("500ns"),
+					},
+				},
+			},
+		},
+		{
+			name: "valid timeout request and backendRequest",
+			rules: []gatewayv1b1.HTTPRouteRule{
+				{
+					Timeouts: &gatewayv1b1.HTTPRouteTimeouts{
+						Request:        toDuration("4s"),
+						BackendRequest: toDuration("2s"),
+					},
+				},
+			},
+		},
+		{
+			name: "valid timeout request",
+			rules: []gatewayv1b1.HTTPRouteRule{
+				{
+					Timeouts: &gatewayv1b1.HTTPRouteTimeouts{
+						Request: toDuration("0s"),
+					},
+				},
+			},
+		},
+		{
+			name:       "invalid timeout request day unit not supported",
+			wantErrors: []string{"Invalid value: \"1d\": spec.rules[0].timeouts.request in body should match '^([0-9]{1,5}(h|m|s|ms)){1,4}$'"},
+			rules: []gatewayv1b1.HTTPRouteRule{
+				{
+					Timeouts: &gatewayv1b1.HTTPRouteTimeouts{
+						Request: toDuration("1d"),
+					},
+				},
+			},
+		},
+		{
+			name:       "invalid timeout request decimal not supported ",
+			wantErrors: []string{"Invalid value: \"0.5s\": spec.rules[0].timeouts.request in body should match '^([0-9]{1,5}(h|m|s|ms)){1,4}$'"},
+			rules: []gatewayv1b1.HTTPRouteRule{
+				{
+					Timeouts: &gatewayv1b1.HTTPRouteTimeouts{
+						Request: toDuration("0.5s"),
+					},
+				},
+			},
+		},
+		{
+			name: "valid timeout request infinite greater than backend request 1ms",
+			rules: []gatewayv1b1.HTTPRouteRule{
+				{
+					Timeouts: &gatewayv1b1.HTTPRouteTimeouts{
+						Request:        toDuration("0s"),
+						BackendRequest: toDuration("1ms"),
+					},
+				},
+			},
+		},
```

Comment:
Can you add a few more tests like this?

1. Request=1s,BackendRequest=200ms (should be valid)
2. Request=10s,BackendRequest=10s (should be valid)
3. Request=200ms,BackendRequest=1s (should not be valid) 

(Also would be great to replicate them in webhook tests while those still exist as well).